File: componet/src/main/java/org/zgl/service/client/hall/HallFriendOperation.java
package org.zgl.service.client.hall;

import org.zgl.dto.clinet.commond.BasePlayerDto;
import org.zgl.dto.clinet.hall.FriendListDto;
import org.zgl.proxy.ClassDesc;
import org.zgl.proxy.MethodDesc;
import org.zgl.proxy.rule.ITcpAsyncService;

/**
 * @author： 猪哥亮
 * @创建时间： 2018/8/17
 * @文件描述：
 */
@ClassDesc("好友")
public interface HallFriendOperation extends ITcpAsyncService {
    boolean addFriend(long targetUid);
    boolean removeFriend(long targetUid);
    @MethodDesc("同意好友添加")
    BasePlayerDto consent(long targetUid);
    @MethodDesc("好友列表")
    FriendListDto friendList();
    @MethodDesc("发送好友聊天消息 msgType:消息类型1-普通文字 2-表情 ，msg：消息内容")
    boolean friendChat(long targetUid, int msgType, String msg);
}

File: cs/proxy/hall/HallFriendOperation.cs
/// <summary>
/// 好友
/// </summary>
namespace org.zgl.service.client.hall{
	[Proxy(gameId = -10)]
	public interface HallFriendOperation : ITcpAsyncService { 
		/// <summary>
		/// 
		/// </summary>
		void addFriend(long targetUid);
		/// <summary>
		///  ----> <<addFriend();回调>>
		/// </summary>
		void addFriend2CallBack(bool callBackParam);
		/// <summary>
		/// 
		/// </summary>
		void removeFriend(long targetUid);
		/// <summary>
		///  ----> <<removeFriend();回调>>
		/// </summary>
		void removeFriend2CallBack(bool callBackParam);
		/// <summary>
		/// 同意好友添加
		/// </summary>
		void consent(long targetUid);
		/// <summary>
		/// 同意好友添加 ----> <<consent();回调>>
		/// </summary>
		void consent2CallBack(BasePlayerDto callBackParam);
		/// <summary>
		/// 好友列表
		/// </summary>
		void friendList();
		/// <summary>
		/// 好友列表 ----> <<friendList();回调>>
		/// </summary>
		void friendList2CallBack(FriendListDto callBackParam);
		/// <summary>
		/// 发送好友聊天消息 msgType:消息类型1-普通文字 2-表情 ，msg：消息内容
		/// </summary>
		void friendChat(long targetUid,int msgType,string msg);
		/// <summary>
		/// 发送好友聊天消息 msgType:消息类型1-普通文字 2-表情 ，msg：消息内容 ----> <<friendChat();回调>>
		/// </summary>
		void friendChat2CallBack(bool callBackParam);
	}
}
